packages/core/src/provider.ts

```diff
@@ -390,3 +409,41 @@ export function addNewChain(
     ]
   })
 }
+
+export async function getPermissions(
+  provider: EIP1193Provider
+): Promise<WalletPermission[]> {
+  try {
+    const permissions = (await provider.request({
+      method: 'wallet_getPermissions'
+    })) as WalletPermission[]
+
+    return Array.isArray(permissions) ? permissions : []
+  } catch (error) {
+    return []
+  }
+}
+
+export async function syncWalletConnectedAccounts(
+  label: WalletState['label']
+): Promise<void> {
+  const wallet = state.get().wallets.find(wallet => wallet.label === label)
```

Comment:
Do we need a guard to confirm `wallet` exists or are we sure it's there at this step?

Reply:
Wallet should exist since this function is only called within the accounts changed listener which gets removed if a wallet is disconnected. There is also a try catch block around the call to this function for graceful failure.